### PR TITLE
localStorage to ls. Switching to storage shim.

### PR DESCRIPTION
**Pull request by bhavul**
Replaces occurances of localStorage to ls.getItem(), ls.setItem() and ls.removeItem().

### CODE REVIEW

File: javascripts/content_scripts/post_new_link.js [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 // Where the Privly URL will be placed is remembered by the contextmenu event
 // or the click event on the button
-var privlyUrlReceiptNode = undefined;
+// var privlyUrlReceiptNode = undefined;    //implicitely all javascript variables are initialized to undefined.
```
Comment (smcgregor) [inline javascripts/content_scripts/post_new_link.js]: I include these to note the scope of the variable. Maybe change this to `var privlyUrlReceiptNode;`

Comment (bhavul) [inline javascripts/content_scripts/post_new_link.js]: All right. I'll make that change.

Comment (bhavul) [inline javascripts/content_scripts/post_new_link.js]: Fixed.

**Comment by smcgregor** [comment]
I'll run the integration tests and merge after the ZeroBin renaming lands (in the next two days).

File: javascripts/background_scripts/first_run.js [hunk 2]
```diff
@@ -64,21 +66,20 @@ var firstRun = {
...
       // the extension installed.
-      if ( localStorage["posting_content_server_url"] !== undefined ) {
-        localStorage["Options:DissableButton"] = "true";
+      if ( ls.getItem("posting_content_server_url") !== undefined ) {
+        ls.setItem("Options:DissableButton","true");
       }
 
-      localStorage["glyph_color"] = Math.floor(Math.random()*16777215).toString(16);
-
+      ls.setItem("glyph_color",Math.floor(Math.random()*16777215).toString(16));
```
Comment (smcgregor) [inline javascripts/background_scripts/first_run.js]: nit: space between each parameter on a function call

File: javascripts/background_scripts/first_run.js [hunk 3]
```diff
@@ -64,21 +66,20 @@ var firstRun = {
...
       var glyph_cells = ((Math.random() < 0.5) ? "false" : "true");
-      for(i = 0; i < 14; i++) {
+      for(var i = 0; i < 14; i++) {
         glyph_cells += "," + ((Math.random() < 0.5) ? "false" : "true");
       }
-      localStorage["glyph_cells"] = glyph_cells;
+      ls.setItem("glyph_cells",glyph_cells);
```
Comment (smcgregor) [inline javascripts/background_scripts/first_run.js]: same as above.

File: javascripts/background_scripts/modal_button.js [hunk 4]
```diff
@@ -14,6 +14,9 @@
  *
  */
 
+
+  /*global chrome:false, ls:true, readingProcess:false, */
```
Comment (smcgregor) [inline javascripts/background_scripts/modal_button.js]: I understand that ls:true puts it into the global namespace, but do the other two directives do?

File: javascripts/content_scripts/post_new_link.js [hunk 5]
```diff
@@ -24,9 +31,6 @@ document.addEventListener( "contextmenu", function(evt) {
   }
 });
 
-// Variable used to indicate whether there is a pending pending operation
-var pendingPost = false;
```
Comment (smcgregor) [inline javascripts/content_scripts/post_new_link.js]: Is this referenced elsewhere? It _was_ in the global scope.

File: javascripts/background_scripts/modal_button.js (hunk 4 above)
Comment (bhavul) [inline javascripts/background_scripts/modal_button.js]: @smcgregor  Yes. true as a property represents that the variable can be edited by the code from current file. False means the code in current file can read that variable but can't make any change to it.

File: javascripts/content_scripts/post_new_link.js (hunk 5 above)
Comment (bhavul) [inline javascripts/content_scripts/post_new_link.js]: Yes. It's used in `javascripts/content_scripts/posting_button.js`. And, it's still in the global scope. Check the commit once again. I've just moved the initialization up.

**Comment by bhavul** [comment]
The above pointed out nits have been fixed. Also, you can go through the comments for the answers to other queries of yours.

**Pull request merged**